src/main/java/org/assertj/core/internal/Strings.java

```diff
@@ -1032,10 +1056,24 @@ public void assertIsSubstringOf(AssertionInfo info, CharSequence actual, CharSeq
    */
   public void assertContainsPattern(AssertionInfo info, CharSequence actual, CharSequence regex) {
     checkRegexIsNotNull(regex);
+    assertContainsPattern(info, actual, Pattern.compile(regex.toString()));
+  }
+
+  /**
+   * Verifies that the given {@code CharSequence} contains the given regular expression.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given {@code CharSequence}.
+   * @param matcher the matcher for finding in the actual {@code CharSequence}.
+   * @throws NullPointerException if the given pattern is {@code null}.
+   * @throws PatternSyntaxException if the regular expression's syntax is invalid.
+   * @throws AssertionError if the given {@code CharSequence} is {@code null}.
+   * @throws AssertionError if the actual {@code CharSequence} does not contain the given regular expression.
+   */
+  public void assertContainsPattern(AssertionInfo info, CharSequence actual, Matcher matcher) {
```

Comment:
add unit tests for this method, look at `Strings_assertIsBase64_Test` for a good example writing tests in AssertJ